### PR TITLE
[BEAM-3373] Add serviceEndpoint parameter to KinesisIO

### PR DESCRIPTION
**Pull request by iemejia**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by iemejia** [comment]
@jbonofre just for reference this parameters allows to enable IT via docker (the emulator case). There will be a different PR by Alexey on mocking for the Unit tests (closest to your suggestion in the JIRA ticket).

**Comment by jbonofre** [comment]
Thanks, I'm taking a look.

**Comment by jbonofre** [comment]
Fail seems due to a quota exceeded. Let me try again, else I will do a clean out.

**Comment by jbonofre** [comment]
retest this please

**Comment by jbonofre** [comment]
I'm testing locally as Jenkins fails with Dataflow runner.
